src/load/locustfile.py

```diff
@@ -1,31 +1,102 @@
 import base64
 
-from locust import HttpLocust, TaskSet, task
+from locust import HttpUser, task, between
 from random import randint, choice
 
 
-class WebTasks(TaskSet):
+class Web(HttpUser):
+    wait_time = between(5, 15)
+
+    @staticmethod
+    def orderUnAcceptable(cart):
+        total = 0
+        for item in cart:
+            total = total + (item["unitPrice"] * item["quantity"])
+        return total > 110
 
     @task
     def load(self):
-        base64string = base64.encodestring('%s:%s' % ('user', 'password')).replace('\n', '')
 
+        # Setup username and password(both are generated)
+        username = "user" + str(randint(0, 1000000))
+        password = str(randint(0, 10000000))
+        base64string = (
+            base64.encodebytes(bytes(("%s:%s" % (username, password)), "utf-8"))
+            .decode()
+            .strip()
+        )
+
+        # Start by visiting MuShop
+        self.client.get("/")
+
+        # Register a user
+        self.client.post(
+            "/api/register",
+            headers={"Content-Type": "application/json"},
+            json={
+                "firstName": "User",
+                "lastName": "Name",
+                "email": "user@example.com",
+                "username": username,
+                "password": password,
+            },
+        )
+
+        # login as the newly registered user
+        self.client.post(
+            "/api/login", headers={"Authorization": "Basic %s" % base64string}
+        )
+
+        # update  profile with shipping address and payment info
+        self.client.post(
+            "/api/address",
+            headers={"Content-Type": "application/json"},
+            json={
+                "number": "000",
+                "street": "000 Example St.",
+                "city": "City",
+                "postcode": "90630",
+                "country": "USA",
+            },
+        )
+        self.client.post(
+            "/api/card",
+            headers={"Content-Type": "application/json"},
+            json={"longNum": "0000000000000000", "expires": "0000", "ccv": "000"},
+        )
+
+        # visit the category page
+        self.client.get("/category.html")
+
+        # get products from the catalogue over the catalogue API, choose random one to buy.
         catalogue = self.client.get("/api/catalogue").json()
         category_item = choice(catalogue)
         item_id = category_item["id"]
 
-        self.client.get("/")
-        self.client.post("/api/login", headers={"Authorization":"Basic %s" % base64string})
-        self.client.get("/category.html")
+        # Visit product page for the chosen product
         self.client.get("/product.html?id={}".format(item_id))
-        self.client.delete("/api/cart")
+
+        # add item to cart
         self.client.post("/api/cart", json={"id": item_id, "quantity": 1})
-        self.client.get("/cart.html")
```

Comment:
Removed the page access to cart? Just api load test?

Reply:
I see the point. I think I'll add this back in and also add a get on the `orders.html`

- I had a few changes I missed like the deployment Yaml - I'll finish updating those and then remove the WIP tag.